### DISCUSSION TITLE
Disable the `clear` calendar button

### DISCUSSION THREAD

**Discussion by madsem** [malzariey/filament-daterangepicker-filter#87] [Q&A]
Hey there,

thanks for this plugin!

After updating to 2.5, there is now this calendar icon to clear the input. But this is not always desired, in our application it makes no sense to clear this and actually could cause some issues.

I couldn't find where this button is even created, is it possible to disable this?

**Comment by malzariey** [comment]
As of now, its not possible to disable it in `DateRangeFilter`

But if you are using the `DateRangePicker` you could override the `suffixAction`

Here is how it is implemented in DateRangePicker, 

```php
 $static->suffixAction(Action::make('clear')
            ->label(__('filament-daterangepicker-filter::message.clear'))
            ->icon('heroicon-m-calendar-days')
            ->action(fn() => $static->clear()));
```

You are welcome to submit a PR to add an API to remove to button to an icon.

**Comment by madsem** [comment]
How about this @malzariey ?
#88
